core/celo_backend.go

```diff
@@ -0,0 +1,49 @@
+package core
+
+import (
+	"context"
+	"errors"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/state"
+	"github.com/ethereum/go-ethereum/core/vm"
+	"github.com/ethereum/go-ethereum/params"
+)
+
+// CeloBackend provide a partial ContractBackend implementation, so that we can
+// access core contracts during block processing.
+type CeloBackend struct {
+	chainConfig *params.ChainConfig
+	state       *state.StateDB
+}
+
+// ContractCaller implementation
+
+func (b *CeloBackend) CodeAt(ctx context.Context, contract common.Address, blockNumber *big.Int) ([]byte, error) {
+	return nil, errors.New("Not implemented")
+}
+
+func (b *CeloBackend) CallContract(ctx context.Context, call ethereum.CallMsg, blockNumber *big.Int) ([]byte, error) {
+	// Ensure message is initialized properly.
```

Comment:
Those seem to come from https://github.com/celo-org/celo-blockchain/blob/b699fd11b08afd7f5e0be00a562813570059f5b3/accounts/abi/bind/backends/simulated.go#L631 , but do you know why they are necessary and why those specific numbers are chosen?

Reply:
Yeah, I got them from the linked place. I assume the value is chosen to be the same as https://github.com/celo-org/op-geth/blob/24c6bfac359e70f5b53bcda47459d7cbefdbc0f0/eth/ethconfig/config.go#L90. We don't have that config available at this place at the moment and I assume any similarly large value would be fine in this place. I didn't really think much about it, yet.